frontend/src/features/public-form/PublicFormProvider.tsx

```diff
@@ -515,32 +574,18 @@ export const PublicFormProvider = ({
         }
       }
 
-      const countryRegionFieldIds = new Set(
-        form.form_fields
-          .filter((field) => field.fieldType === BasicField.CountryRegion)
-          .map((field) => field._id),
-      )
-      // We want users to see the country/region options in title-case but we also need the data in the backend to remain in upper-case.
-      // Country/region data in the backend needs to remain in upper-case so that they remain consistent with myinfo-countries.
-      const formInputsWithCountryRegionInUpperCase = Object.keys(
-        formInputs,
-      ).reduce((newFormInputs: typeof formInputs, fieldId) => {
-        const currentInput = formInputs[fieldId]
-        if (
-          countryRegionFieldIds.has(fieldId) &&
-          typeof currentInput === 'string'
-        ) {
-          newFormInputs[fieldId] = currentInput.toUpperCase()
-        } else {
-          newFormInputs[fieldId] = currentInput
-        }
-        return newFormInputs
-      }, {})
+      const transformedFormInputs = transformFormInputTrimTextInputs(
```

Comment:
/nit I initially missed out that there's a nested function call. Wondering if this be made more explicit?

```tsx
let transformedFormInputs = formInput
transformedFormInputs = transformFormInputCountryRegionToUpperCase(
  transformedFormInputs,
  form.form_fields
)
transformedFormInputs = transformFormInputTrimTextInputs(
  transformedFormInputs,
  form.form_fields
)
```
vs 
```tsx
const transformedFormInputs = transformFormInputTrimTextInputs(
  transformFormInputCountryRegionToUpperCase(
    formInputs,
    form.form_fields,
  ),
  form.form_fields,
) as typeof formInputs
```

Reply:
Looks v. neat!